src/Consolonia.PlatformSupport/GpmConsole.cs

```diff
@@ -0,0 +1,301 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Avalonia;
+using Avalonia.Input;
+using Avalonia.Input.Raw;
+using Avalonia.Threading;
+using Consolonia.Core.Helpers;
+using Consolonia.Core.Infrastructure;
+
+namespace Consolonia.PlatformSupport
+{
+    /// <summary>
+    ///     Console implementation with GPM (General Purpose Mouse) support for TTY environments
+    ///     This wraps CursesConsole and adds libgpm mouse input handling
+    /// </summary>
+    public class GpmConsole : CursesConsole
+    {
+        private readonly CancellationTokenSource _gpmCancellation;
+        private GpmConnect _gpmConnection;
+        private int _gpmFd = -1;
+        private bool _gpmInitialized;
+
+        public GpmConsole()
+            : base(false)
+        {
+            _gpmCancellation = new CancellationTokenSource();
+            InitializeGpm();
+        }
+
+
+        private void InitializeGpm()
+        {
+            try
+            {
+                // Set up GPM connection
+                _gpmConnection = new GpmConnect
+                {
+                    EventMask = 0xffff, // Receive all events
+                    DefaultMask = 0, // Explicitly disable all default handling
+                    MinMod = 0, // Accept events with no modifiers or more
+                    MaxMod = 0xffff // Accept events with any/all modifiers (0xFFFF or ~0)
+                };
+
+                _gpmFd = Gpm.Open(ref _gpmConnection, 0);
+                if (_gpmFd < 0) return;
+
+                // Hide the GPM hardware cursor (we draw our own in software)
+                try
+                {
+                    _ = Gpm.DrawPointer(-1, -1, 0);
+                }
+                catch (EntryPointNotFoundException)
+                {
+                    // Function not available, cursor will remain visible
+                    Debug.WriteLine("Gpm_DrawPointer not available, GPM cursor will be visible");
+                }
+
+                _gpmInitialized = true;
+                Task.Run(GpmEventLoop, _gpmCancellation.Token);
+            }
+            catch (DllNotFoundException)
+            {
+            }
+            catch (EntryPointNotFoundException)
+            {
+            }
+        }
+
+        private async Task GpmEventLoop()
+        {
+            await Helper.WaitDispatcherInitialized();
+
+            while (!_gpmCancellation.Token.IsCancellationRequested && !Disposed)
```

Comment:
I'm almost sure left part of condition can throw ObjectDisposedException

Reply:
updated